src/talk/renderer/Viewer/ViewerHandlerBase.vue

```diff
@@ -0,0 +1,91 @@
+<!--
+  - SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+  -->
+
+<script setup>
+import { computed } from 'vue'
+import { t } from '@nextcloud/l10n'
+import NcEmptyContent from '@nextcloud/vue/dist/Components/NcEmptyContent.js'
+import NcLoadingIcon from '@nextcloud/vue/dist/Components/NcLoadingIcon.js'
+import IconAlertCircleOutline from 'vue-material-design-icons/AlertCircleOutline.vue'
+
+const props = defineProps({
+	loading: {
+		type: Boolean,
+		required: false,
+		default: false,
+	},
+	error: {
+		type: [String, Boolean],
+		required: false,
+		default: false,
+	},
+	/**
+	 * In lazy mode the content is rendered visually hidden during loading.
+	 * Used when loading happens on rendering, for example, loading <img>.
+	 */
+	lazy: {
+		type: Boolean,
+		required: false,
+	},
+})
+
+const readyToShow = computed(() => !props.loading && !props.error)
+</script>
+
+<template>
+	<div class="viewer-wrapper">
+		<NcEmptyContent v-if="loading" :name="t('talk_desktop', 'Loading …')" class="delayed-appear">
+			<template #icon>
+				<NcLoadingIcon />
+			</template>
+		</NcEmptyContent>
+
+		<NcEmptyContent v-else-if="error"
+			:name="t('talk_desktop', 'Could not load the file')"
+			:description="typeof error === 'string' ? error : undefined">
+			<template #icon>
+				<IconAlertCircleOutline />
+			</template>
+		</NcEmptyContent>
+
+		<div v-if="lazy || readyToShow" class="viewer-wrapper__content" :class="{ 'hidden-visually': lazy && !readyToShow }">
+			<slot />
+		</div>
+	</div>
+</template>
+
+<style scoped>
+.viewer-wrapper {
+	display: flex;
+	justify-content: center;
+	align-items: center;
+	height: 100%;
+}
+
+.viewer-wrapper__content {
+	width: 100%;
+	height: 100%;
+}
+
+.viewer-wrapper__content > :deep(*) {
+	width: 100%;
+	height: 100%;
+}
+
+@keyframes delayed-appear {
+	0% {
+		opacity: 0;
+	}
+	100% {
+		opacity: 1;
+	}
+}
+
+.delayed-appear {
```

Comment:
To not blink. If loading is quite fast, it is better to show the result in 0.2-0.3s (slightly noticeable). You just click and have a file opened. Rather than to blink with loading spinner for just 0.2s.